Source/items.cpp

```diff
@@ -4477,13 +4456,13 @@ void SpawnSmith(int lvl)
 			memset(&Items[0], 0, sizeof(*Items));
 			Items[0]._iSeed = AdvanceRndSeed();
 			SetRndSeed(Items[0]._iSeed);
-			int idata = RndSmithItem(lvl) - 1;
-			GetItemAttrs(0, idata, lvl);
+			int itemData = RndSmithItem(lvl) - 1;
+			GetItemAttrs(Items[0], itemData, lvl);
 		} while (Items[0]._iIvalue > maxValue);
 		smithitem[i] = Items[0];

```

Comment:
```suggestion
			GetItemAttrs(smithitem[i], itemData, lvl);
		} while (smithitem[i]._iIvalue > maxValue);
```

No need to use `Items[0]` as a temp buffer anymore :)